R/learner_abess_classif_abess.R

```diff
@@ -0,0 +1,157 @@
+          tags = "train"
+        ),
+        tune.type = p_fct(
+          default = "gic",
+          levels = c("gic", "aic", "bic", "ebic", "cv"),
+          tags = "train"
+        ),
+        normalize = p_int(default = NULL, special_vals = list(NULL), tags = "train"),
+        support.size = p_uty(default = NULL, tags = "train"),
+        support.size.predict = p_int(default = NULL, lower = 1 , special_vals = list(NULL), tags = "train"),
+        c.max = p_int(default = 2, lower = 1, tags = "train"),
+        gs.range = p_uty(default = NULL, tags = "train"),
+        lambda = p_dbl(default = 0.0, lower = 0.0, tags = "train"),
+        always.include = p_uty(default = NULL, tags = "train"),
+        group.index = p_uty(default = NULL, tags = "train"),
+        init.active.set = p_uty(default = NULL, tags = "train"),
+        splicing.type = p_int(default = 2, lower = 1, upper = 2, tags = "train"),
+        max.splicing.iter = p_int(default = 20, lower = 1, tags = "train"),
+        screening.num = p_int(default = NULL, lower = 0, special_vals = list(NULL), tags = "train"),
+        important.search = p_int(default = NULL, lower = 0, special_vals = list(NULL), tags = "train"),
+        warm.start = p_lgl(default = TRUE, tags = "train"),
+        nfolds = p_int(default = 5,  tags = "train"),
+        foldid = p_uty(default = NULL, tags = "train"),
+        cov.update = p_lgl(default = FALSE, tags = "train"),
+        newton = p_fct(default = "exact", levels = c("exact", "approx"), tags = "train"),
+        newton.thresh = p_dbl(default = 1e-6, lower = 0.0, tags = "train"),
+        max.newton.iter = p_int(default = NULL, lower = 1, special_vals = list(NULL), tags = "train"),
+        early.stop = p_lgl(default = FALSE, tags = "train"),
+        ic.scale = p_dbl(default = 1.0, lower = 0.0, tags = "train"),
+        num.threads = p_int(default = 0L, lower = 0L, tags = c("train", "threads")),
+        seed = p_int(default = 1, tags = "train")
+      )
+
+      super$initialize(
+        id = "classif.abess",
+        packages = "abess",
+        feature_types = c("numeric", "integer"),
+        predict_types = c("response", "prob"),
+        param_set = param_set,
+        properties = c("weights", "selected_features", "twoclass", "multiclass"),
+        man = "mlr3extralearners::mlr_learners_classif.abess",
+        label = "Fast Best Subset Selection for Classification"
+      )
+    },
+    #' @description
+    #' Extract the name of selected features from the model by [abess::extract()].
+    #' @param support.size an integer which indicates the number of selected features,
+    #'        default is selected automatically by abess algorithm.
+    #' @return The name of selected features
+    selected_features = function(support.size = NULL) {
+      abess::extract(self$model, support.size = support.size)$support.vars
+    }
+  ),
+  private = list(
+    .train = function(task) {
+      self$param_set$values$num.threads = 1L
```

Comment:
don't modify the actual param set in-place. 
If you want to do this, use `pv = self$param_set$get_values(tags = "train")` and `pv$values$num.threads = 1L`